### PR TITLE
Fix frame dupe raising TypeError

### PR DESCRIPTION
**Pull request by RobLoach**
`CompositeEnvironmentDriver.video_refresh` matches `data.value` against `case 0:` to detect a frame dupe, but ctypes reports a NULL `c_void_p`'s `value` as `None`, so that branch is unreachable and a core passing NULL to `retro_video_refresh_t` falls through to the `case _:` `TypeError` — raised out of the core's own callback.

Using `None | 0` fixes it.

### CODE REVIEW

**Comment by JesseTG** [comment]
Thanks for contributing! Could you retarget this against `dev`?

**Comment by RobLoach** [comment]
Sounds good. Think we need the test at all? Since its just a None 0 thing, I would be fine to live without it. Your call.

**Review by JesseTG — APPROVED** [review]

**Review by JesseTG — APPROVED** [review]

**Comment by JesseTG** [comment]
I've approved your PR to run workflows, but it looks like doing so uncovered some typing issues. Could you address them before I merge this? Don't forget to pull from `main`, I've pushed a new release (which also removes a `@final` that's triggering an error).

File: tests/unit/drivers/test_composite_video_refresh.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    """The ordinary case: a pointer to pixels arrives as a view of those pixels."""
+    pixels = bytearray(range(PITCH * HEIGHT))
+    block = (c_char * len(pixels)).from_buffer(pixels)
+
+    env.video_refresh(c_void_ptr(addressof(block)), WIDTH, HEIGHT, PITCH)
+
+    assert len(video.frames) == 1
+    frame = video.frames[0]
+    assert isinstance(frame, memoryview)
+    assert bytes(frame) == bytes(pixels)
```
Comment (RobLoach) [inline tests/unit/drivers/test_composite_video_refresh.py:85]: Cool. Again, I don't believe the tests here are needed, but feel free to have them if you enjoy testy tests.

``` suggestion
```

**Comment by JesseTG** [comment]
All good, including the test (better safe than sorry). Just retarget it against `dev` then I'll merge it.
